chalicelib/app_utils.py

```diff
@@ -771,6 +771,47 @@ def run_get_environment(environ):
     return process_response(response)
 
 
+def run_delete_environment(environ):
+    """
+    Removes the environ entry from the Foursight envs bucket. This effectively de-schedules all checks
+    but does not remove any data.
+    """
+    s3_connection = S3Connection('foursight-envs')
+    s3_resp = s3_connection.delete_keys([environ])
+    keys_deleted = s3_resp['Deleted']
+    if not keys_deleted:
+        response = Response(
+            body = {
+                'status': 'error',
+                'description': 'Invalid environment provided to delete',
+                'environment': environ
+            },
+            status_code = 400
+        )
+    else:
+        our_key = keys_deleted[0]  # since we only passed on key to be deleted, the response will be a length 1 list
+        if our_key['Key'] != environ:
+            response = Response(
+                body = {
+                    'status': 'error',
+                    'description': 'An error occured during environment deletion, please check S3 directly',
```

Comment:
typo: 'occurred' has two r's. 

But also, is there any security danger in revealing the nature of the error in a more clear way?
```
'description': 'An error occurred during environment deletion: the environment named in the response (%s) did not match the environment given (%s). Please check S3 directly' % (our_key['Key'], environ),
```
If the `Key` is potentially a private object, it's OK to leave out the parentheticals in the above and just say that the object returned didn't match.

Finally, is `our_key` really the right name for this result? At minimum you don't know if it's ours until you check it matches, but even beyond that, is the object out of which `'Key'` is taken itself just a structured representation of a key?

Reply:
`our_key` is okay IMO because it should never be anything else except a length 1 list holding the key we requested. If it returns something else, that would mean a problem occurred on the AWS side (not due to us) - which we would like to catch if it does somehow happen.